ci/main.py

```diff
@@ -3,61 +3,130 @@
 import json
 import yaml
 import glob
-from subprocess import check_call, check_output, CalledProcessError
+import logging
+from subprocess import check_call, check_output
 
 from conda_build.config import config
 
 
-CHANNEL = 'https://conda.anaconda.org/biocore'
+log = logging.getLogger(__name__)
+handler = logging.StreamHandler(sys.stdout)
+handler.setFormatter(
+    logging.Formatter('%(asctime)s %(levelname)s: %(message)s'))
+handler.setLevel(logging.INFO)
+log.addHandler(handler)
+log.setLevel(logging.INFO)
 
 
-def build_upload_recipes(p):
+def build_upload_recipes(p, channel):
+    '''Build & upload recipes recursively in a directory.
+
+    Parameters
+    ----------
+    p : str
+        Path to the recipes.
+    channel : str
+        Anaconda channel where the packages will be uploaded.
+    '''
     for root, dirs, files in os.walk(p):
         has_recipe = 'meta.yaml' in files
         if not dirs and has_recipe:
             with open(os.path.join(root, 'meta.yaml')) as f:
-                pkg = yaml.load(f)['package']
```

Comment:
add to a log of what recipe is being checked? to know what recipe is being read in case weird things happen(such as a recipe dir is skipped or stopped before checking if uploaded)